packages/react-instantsearch-core/src/connectors/__tests__/connectQueryRules.ts

```diff
@@ -0,0 +1,122 @@
+import connect from '../connectQueryRules';
+
+jest.mock('../../core/createConnector', () => (component: React.Component) =>
+  component
+);
+
+describe('connectQueryRules', () => {
+  describe('single index', () => {
+    const indexName = 'index';
+    const context = { context: { ais: { mainTargetedIndex: indexName } } };
+    const getProvidedProps = connect.getProvidedProps.bind(context);
+
+    describe('without userData', () => {
+      it('provides the correct props to the component', () => {
+        const props = {};
+        const searchState = {};
+        const searchResults = {
+          results: { [indexName]: { userData: undefined } },
+        };
+
+        expect(getProvidedProps(props, searchState, searchResults)).toEqual({
+          items: [],
+          canRefine: false,
+        });
+      });
+    });
+
+    describe('with userData', () => {
+      it('provides the correct props to the component', () => {
+        const props = {};
+        const searchState = {};
+        const searchResults = {
+          results: {
+            [indexName]: { userData: [{ banner: 'image.png' }] },
+          },
+        };
+
+        expect(getProvidedProps(props, searchState, searchResults)).toEqual({
+          items: [{ banner: 'image.png' }],
+          canRefine: true,
+        });
+      });
+
+      it('transforms items before passing the props to the component', () => {
+        const props = {
+          transformItems: () => [{ banner: 'image-transformed.png' }],
+        };
+        const searchState = {};
+        const searchResults = {
+          results: {
+            [indexName]: { userData: [{ banner: 'image.png' }] },
+          },
+        };
+
+        expect(getProvidedProps(props, searchState, searchResults)).toEqual({
+          items: [{ banner: 'image-transformed.png' }],
+          canRefine: true,
+        });
+      });
+    });
+  });
+
+  describe('multi index', () => {
```

Comment:
I'm not sure if the "multi index" test suit makes sense for this connector. This is the same as the "single index" test suit, with a different context.

Reply:
Still makes sense to test that it's actually calling `getResults` correctly no?

Reply:
We can keep it, the multi-index is a feature that does a lot of work behind the hood. But you can test on a different index that the main one, otherwise yes it's a bit useless.